scripts/integration/integration.R

```diff
@@ -142,51 +142,82 @@ runHarm = function(sobj, batch) {
       return(sobj)
 }
 
-runLiger = function(sobj, batch, hvg, k=20, res=0.4, small.clust.thresh=20) {
-    require(liger)
-    require(Seurat)
-
-    # Only counts is converted to liger object. To pass our own normalized data,
-    # store it in the "counts" slot
+runLiger = function(sobj, batch, hvg, k = 20, res = 0.4, small.clust.thresh = 20) {
+  tryCatch(
+    require(liger),
+    warning = function (w) require(rliger)
+  )
+  require(Seurat)
+
+  # Only counts is converted to liger object. To pass our own normalized data,
+  # store it in the "counts" slot
+  if (is.null(sobj@assays$RNA)) {
+    # Seurat v4
+    data <- GetAssayData(sobj, slot = "data")
+    SetAssayData(sobj, slot = "counts", new.data = as.matrix(data))
```

Comment:
you pin seurat to v3 above, but then have this flexible assay assignment here. Why pin it then?

Reply:
or is this for seurat v4 data objects?

Reply:
I started adapting for Seurat v4, and got it working for harmony, but gave up for other methods. Moving to Seurat v4 should be a task for another pull request

Reply:
ok